wgsl/index.bs

```diff
@@ -9346,6 +9346,60 @@ struct __modf_result_vecN {
         [=Component-wise=] when |T| is a vector.
         (SPIR-V OpBitCount)
 
+  <tr algorithm="signed extract bits">
+    <td>|T| is [SIGNEDINTEGRAL]
+    <td class="nowrap">`extractBits(`|e|`:` |T| `, `|offset|`: u32, `|count|` : u32) ->` |T|
+    <td>Reads bits from an integer, with sign extension.
+
+    When |T| is a scalar type, then:
+    <ul>
+    <li>|w| is the bit width of |T|
+    <li>|o| = min(|offset|,|w|)
+    <li>|c| = min(|count|, |w| - |o|)
+    <li>The result is 0 if |c| is 0.
+    <li>Otherwise, bits 0..|c|-1 of the result are copied from bits
+       |o|..|o|+|c|-1 of |e|.
+       Other bits of the result are the same as bit |c|-1 of the result.
+    </ul>
+    [=Component-wise=] when |T| is a vector.
+    <br>(OpBitFieldSExtract)
+
+  <tr algorithm="unsigned extract bits">
+    <td>|T| is [UNSIGNEDINTEGRAL]
+    <td class="nowrap">`extractBits(`|e|`:` |T| `, `|offset|`: u32, `|count|` : u32) ->` |T|
+    <td>Reads bits from an integer, without sign extension.
+
+    When |T| is a scalar type, then:
+    <ul>
+    <li>|w| is the bit width of |T|
+    <li>|o| = min(|offset|,|w|)
+    <li>|c| = min(|count|, |w| - |o|)
+    <li>The result is 0 if |c| is 0.
+    <li>Otherwise, bits 0..|c|-1 of the result are copied from bits
+       |o|..|o|+|c|-1 of |e|.
+       Other bits of the result are 0.
+    </ul>
+    [=Component-wise=] when |T| is a vector.
+    <br>(OpBitFieldUExtract)
+
+  <tr algorithm="insert bits">
+    <td>|T| is [UNSIGNEDINTEGRAL]
```

Comment:
Shouldn't `T` be integral (both u32 and i32)?

Reply:
fixed.